auth.go

```diff
@@ -530,14 +557,19 @@ func (x *Central) RenameIdentity(oldIdent, newIdent string) error {
 
 	x.Log.Infof("RenameIdentity (PermitDB) successful (%v -> %v)", oldIdent, newIdent)
 
-	eInvalidate := x.InvalidateSessionsForIdentity(oldIdent)
-	x.Log.Infof("RenameIdentity (%v -> %v), session invalidation result (%v)", oldIdent, newIdent, eInvalidate)
-	return nil
+	if userId, eGetUserId := x.userstore.GetUserIdFromIdentity(newIdent); eGetUserId == nil {
```

Comment:
It is very important that a multistep process such as this, which is exposed as an apparently atomic API, tries it's best to be atomic. It's very bad if two parts of the process succeed, and the third part fails. I'm pretty sure that GetUserIdFromIdentity should be the first thing happening here. Also, we need to be checking whether the new identity is already in use, and failing with an appropriate error code if that is the case.